ontopy/ontology.py

```diff
@@ -424,20 +428,28 @@ def get_by_label_all(self, label, label_annotations=None, prefix=None):
                 f"Invalid label definition, {label!r} contains spaces."
             )
 
-        if label_annotations:
-            annotations = (
-                ann.name if hasattr(ann, "storid") else ann
-                for ann in label_annotations
-            )
-        elif self._label_annotations:
-            annotations = (ann.name for ann in self.label_annotations)
+        if label_annotations is None:
+            label_annotations = self.label_annotations
 
-        else:
-            annotations = None
         entities = set()
-        if annotations:
-            for key in annotations:
-                entities.update(self.world.search(**{key: label}))
+        for storid in self._to_storids(label_annotations):
+            label_entity = self._unabbreviate(storid)
+            key = (
+                label_entity.name
+                if hasattr(label_entity, "name")
+                else label_entity
+            )
+            entities.update(self.world.search(**{key: label}))
+
+        # ALTERNATIVE IMPLEMENTATION THAT DOESN'T INVOLVE search().
+        # IT IS FASTER AND MIGHT AVOID SURPRICES.  SHOULD WE ADD AN
+        # OPTION TO CHOOSE BETWEEN SEARCH() AND THIS IMPLEMENTATION?
+        # entities = set()
+        # for storid in self._to_storids(label_annotations):
+        #    entities.update(
+        #        self.world[self._unabbreviate(s)]
+        #        for s, _, _, _ in get_triples(None, storid, label, None)
+        #    )
```

Comment:
Why is it faster? If the outcome is exactly the same in both versions I do not see the reason to have two ways of doing the same thing. Then  think we should go for the fastest solution. Did you time it?

Reply:
Added exact_match argument as you suggested.